compiler/optimizer/LocalLiveRangeReducer.cpp

```diff
@@ -370,6 +370,21 @@ void TR_LocalLiveRangeReduction::populatePotentialDeps(TR_TreeRefInfo *treeRefIn
                }
             }
 
+         // If there are store nodes before arraycmp, arraycmp can share base addresses
+         // with them. UseonlyAliases needs to be set up to prevent the
+         // store nodes from being moved beyond arraycmp.
+         if ((opCode.getOpCodeValue() == TR::arraycmp) && !symRef->getUseonlyAliases().isZero(comp()))
```

Comment:
Is `arraycmp` the only opcode that could run into this situation with use-only aliases?

Reply:
The existing implementation in [TR_LocalLiveRangeReduction::populatePotentialDeps() ](https://github.com/eclipse/omr/blob/cd8a2e017030701fc5b39ea162acdec128c6aecb/compiler/optimizer/LocalLiveRangeReducer.cpp#L361) considers only UseDefAliases.  `arraycmp` [doesn't have any UseDefAliases](https://github.com/eclipse/omr/blob/cd8a2e017030701fc5b39ea162acdec128c6aecb/compiler/il/Aliases.cpp#L347-L348), it ends up not setting up any sym dependency for `arraycmp` in populatePotentialDeps. Theoretically speaking, I guess it's possible that the issue could happen to other opcode, but `arraycmp` is the only case I can confirm that have the issue because of the missing dependency here.  I'm not sure why historically in populatePotentialDeps it doesn't check both UseDefAliases and UseonlyAliases as below. Maybe @vijaysun-omr could shed some light. Should the limit on `arraycmp` be removed as below and check UseonlyAliases in [all cases](https://github.com/eclipse/omr/blob/cd8a2e017030701fc5b39ea162acdec128c6aecb/compiler/optimizer/LocalLiveRangeReducer.cpp#L354) in populatePotentialDeps? 

```c++
         if (!symRef->getUseDefAliases(isCallDirect).isZero(comp()))
            {
           ... 
            }

         if (!symRef->getUseonlyAliases().isZero(comp()))
            {
           ... 
            }
```

Reply:
I feel it is safer to do this for other opcodes too. However you may need to run a couple of the usual performance tests to be sure that it does not harm throughput (I doubt that it will).

Reply:
I ran ILOG/ODM and DT7 with the baseline and this fix. Here is the result. I don't see this fix show performance impact. 

|  | Baseline | Fix | Delta |
| ------------- | ------------- |------------- |------------- |
| ILOG/ODM 4way-Seg5FastpathRVEJB | 5984 | 6051 | 1.01 |
|ILOG/ODM 4way-Seg300RulesFastpathRVEJB | 157187 | 157032 | 0.999 |

|  | Baseline | Fix | Delta |
| ------------- | ------------- |------------- |------------- |
| DT7 17dev-4way-LargeThreadPool  | 4046 | 4069 | 1.01 |
| DT7 17dev-4way-LargeThreadPoolwarm | 4114 | 4101 | 0.997 |

Reply:
Thanks for checking, that looks good to me @a7ehuo

Reply:
The fix is updated in 6f696af  to apply the check on UseonlyAliases for all opcodes instead of only arraycmp. Ready for another review. Thanks!